File: c/src/Structural/Bridge/supersizesoda.c


#include "stdio.h"

//SuperSizeSoda - two of two classes extending the Abstract

void supersizesoda_pourSoda() 
{
	int i;
	void (*pourSodaImp)() = SodaImp_get();
   for (i = 0; i < 5; i++) 
	{
		printf("...glug...\n");
		pourSodaImp();
	}
}

File: php/src/Structural/Bridge/SuperSizeSoda.php
<?

#//SuperSizeSoda.java - two of two classes extending the Abstract

require_once 'Soda.php';

class SuperSizeSoda extends Soda
{

	function __construct()
	{
		$this->setSodaImp();
	}

	function pourSoda ()
	{
		$sodaImp = $this->getSodaImp();
		$retval;
		for($i=0; $i < 5; $i++)
			$retval[]= $sodaImp->pourSodaImp();
		return implode(' ',$retval);
	}
}
